backend/main_test.go

```diff
@@ -476,3 +465,80 @@ func TestImproveBadMethods(t *testing.T) {
 	postresp := makeHTTPRequest(postreq, handler, t)
 	checkResponseStatus(http.StatusMethodNotAllowed, postresp, t)
 }
+
+type failingAuthStub struct{}
+
+func (auth failingAuthStub) authenticate(next http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		http.Error(w, "Auth failed", http.StatusUnauthorized)
+		return
+	}
+}
+
+func TestAuthMiddleware(t *testing.T) {
+	server := Server{store: in_memory_storage.MakeInMemStore()}
+	server.auth = failingAuthStub{}
+	handler := server.makeHandler()
+
+	testHandlerWithAuth := func(httpMethod, target string, body io.Reader) {
```

Comment:
Ideally the name would be a bit clearer as to what this function is actually doing. What it's doing is calling the supplied HTTP endpoint and asserting that we get a 401 error.

`assertAuthenticationFailure` maybe?

Reply:
Yes, I definitely struggled to find an expressive, readable name for this, I've changed it to your suggestion.